### PR TITLE
adding end trailers

### PR DESCRIPTION
**Pull request by jayanth-tatina-groww**
- Enabling HTTP2 in Gin Server
- Using HTTP2 client to connect to upstream servers
- Adding End trailers which are passed by GRPC Clients

### CODE REVIEW

File: router/gin/router.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (r ginRouter) Run(cfg config.ServiceConfig) {
 	go r.cfg.Engine.Run("XXXX")
 
 	r.cfg.Logger.Info("[SERVICE: Gin] Listening on port:", cfg.Port)
-	if err := r.runServerF(r.ctx, cfg, r.cfg.Engine); err != nil && err != http.ErrServerClosed {
+	if err := r.runServerF(r.ctx, cfg, r.cfg.Engine.Handler()); err != nil && err != http.ErrServerClosed {
```
Comment (kamal-groww) [inline router/gin/router.go:110]: Why is this change required ?

**Review by kamal-groww — APPROVED** [review]
Looks good

File: router/gin/router.go (hunk 1 above)
Comment (jayanth-tatina-groww) [reply router/gin/router.go:110]: for gin to set the http2 handler this function has to be called 
```
func (engine *Engine) Handler() http.Handler {
	if !engine.UseH2C {
		return engine
	}

	h2s := &http2.Server{}
	return h2c.NewHandler(engine, h2s)
}
```
